Source/DafnyCore/Compilers/Python/Compiler-python.cs

```diff
@@ -177,20 +177,29 @@ class PythonCompiler : SinglePassCompiler {
       var relevantTypeParameters = typeParameters.Where(NeedsTypeDescriptor);
       var args = relevantTypeParameters.Comma(tp => tp.GetCompileName(Options));
       if (!string.IsNullOrEmpty(args)) { args = $", {args}"; }
+      var isNewtypeWithTraits = cls is NewtypeDecl { ParentTraits: { Count: > 0 } };
+      if (isNewtypeWithTraits) {
+        args += ", value";
+      }
       var block = methodWriter.NewBlockPy(header: $"def  __init__(self{args}):", close: BlockStyle.Newline);
       foreach (var tp in relevantTypeParameters) {
         block.WriteLine("self.{0} = {0}", tp.GetCompileName(Options));
       }
       var constructorWriter = block.Fork();
       block.WriteLine("pass");
 
-      methodWriter.NewBlockPy("def __dafnystr__(self) -> str:")
-        .WriteLine($"return \"{fullPrintName}\"");
+      var wBody = methodWriter.NewBlockPy("def __dafnystr__(self) -> str:");
+      if (isNewtypeWithTraits) {
+        wBody.WriteLine("return str(self._value)");
```

Comment:
This should be:
```suggestion
        wBody.WriteLine("return _dafny.string_of(self._value)");
```
but it seems dead to me:
```Dafny
trait Trait { }

newtype MyInt extends Trait = real

method Main() {
  var m: MyInt := 1.0/3.0;
  print m, "\n";
}
```

Reply:
The problem is actually unrelated. You generate the wrong division.

Reply:
I made the `_dafny.string_of` change.
I don't understand what it is you say I'm generating into the wrong division. Can you give more details?

Reply:
The result should be 1/3, but we print 0 instead, because we use integer division.

Reply:
Yes, I noticed that, too. I have a fix, but I'll include it in a separate PR.

Reply:
The 1/3 issue is fixed in https://github.com/dafny-lang/dafny/pull/4234